docs/authoring-workspaces.md

```diff
@@ -0,0 +1,107 @@
+
+**Use [the vanilla workspace bundle](../workspaces/vanilla/README.md) and [others](../workspaces/) as reference** or as the basis for your new bundle. To start from scratch follow the Porter [Quickstart Guide](https://porter.sh/quickstart/) ([`porter create` CLI command](https://porter.sh/cli/porter_create/) will generate a new bundle in the current directory).
+
+## Prerequisites
+
+* [Docker installed](https://docs.docker.com/get-docker/)
+* [Porter installed](https://porter.sh/install)
+* Azure TRE instance deployed to test against
+
+## Workspace bundle manifest
+
+The manifest of a workspace bundle is the `porter.yaml` file (see [Author Bundles in Porter documentation](https://porter.sh/author-bundles/)). This section describes the mandatory credentials, input and output parameters of a TRE workspace bundle.
+
+### Credentials
+
+A workspace bundle requires the following [credentials](https://porter.sh/author-bundles/#credentials) to provision resources in Azure:
+
+* [Azure tenant ID](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-how-to-find-tenant)
+* Azure subscription ID
+* The client ID of a [service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals) with privileges to provision resources
+* The client secret (password) of a service principal
+
+The credentials are provided as environment variables by the deployment runner. The bundle author must use the following environment variable names:
+
+* `AZURE_TENANT_ID`
+* `AZURE_SUBSCRIPTION_ID`
+* `AZURE_SP_CLIENT_ID`
+* `AZURE_SP_PASSWORD`
+
+The names of the Porter credentials (`name` field in `porter.yaml`) can be freely chosen by the author.
+
+Example:
+
+```yaml
+credentials:
+- name: azure_tenant_id
+  env: AZURE_TENANT_ID
+- name: azure_subscription_id
+  env: AZURE_SUBSCRIPTION_ID
+- name: azure_service_principal_client_id
+  env: AZURE_SP_CLIENT_ID
+- name: azure_service_principal_password
+  env: AZURE_SP_PASSWORD
+```
+
+### Parameters
+
+This section describes the mandatory [(input) parameters](https://porter.sh/author-bundles/#parameters) of a workspace bundle manifest.
+
+| Parameter | Type | Description | Example value |
+| --------- | ---- | ----------- | ------------- |
+| `core_id` | string | The ID of the parent TRE instance. | `mytre-dev-3142` |
+| `workspace_id` | string | Unique 4-digit workspace ID. | `2718` |
+| `location` | string | Azure location (region) to deploy the workspace resource to. | `westeurope` |
+| `address_space` | string | VNet address space for the workspace services. | `10.2.1.0/24` |
+
+"Core" in `core_id` refers to the Azure TRE instance and its base (core) infrastrure. `core_id` can be found in the resource names of the Azure TRE instance; for example the resource group name of the Azure TRE instance based on the example in the above table would be "`rg-mytre-dev-3142`".
```

Comment:
Should we describe it as resource group?

Reply:
Describe what as resource group?

Reply:
Let's address this in the future revision of the document.